app/models/merchant.rb

```diff
@@ -8,15 +8,20 @@ class Merchant < ApplicationRecord
   validates :name, presence: true
 
   def top_five_customers
-    self.transactions.joins(invoice: :customer)
-      .where("transactions.result = ?", "0")
-      .select("customers.id, CONCAT(customers.first_name,' ', customers.last_name) as full_name, COUNT(transactions) as successful_transactions")
-      .group("customers.id")
-      .order("successful_transactions DESC")
+    Customer.select("customers.*, CONCAT(customers.first_name, ' ', customers.last_name) AS full_name, count(distinct transactions.id) as successful_transactions")
+      .joins(invoices: [:transactions, invoice_items: {item: :merchant}])
```

Comment:
A full name model method was created for one of the admin user stories if we want to look at implementing that later on.